rskj-core/src/test/java/co/rsk/peg/union/UnionStorageProviderImplTest.java

```diff
@@ -0,0 +1,187 @@
+        // Arrange
+        storageAccessor.saveToRepository(
+            UnionStorageIndexKey.UNION_BRIDGE_CONTRACT_ADDRESS.getKey(),
+            rskAddress,
+            BridgeSerializationUtils::serializeRskAddress
+        );
+
+        // Act
+        Optional<RskAddress> unionBridgeAddress = unionStorageProvider.getUnionAddress();
+
+        // Assert
+        assertTrue(unionBridgeAddress.isPresent());
+        assertEquals(rskAddress, unionBridgeAddress.get());
+    }
+
+    @ParameterizedTest
+    @MethodSource("validAddressProvider")
+    void getUnionAddress_whenAddressSet_shouldReturnCacheAddress(RskAddress rskAddress) {
+        // Arrange
+        unionStorageProvider.setUnionAddress(rskAddress);
+
+        // Act
+        Optional<RskAddress> unionBridgeAddress = unionStorageProvider.getUnionAddress();
+
+        // Assert
+        assertTrue(unionBridgeAddress.isPresent());
+        assertEquals(rskAddress, unionBridgeAddress.get());
+    }
+
+    @ParameterizedTest
+    @MethodSource("validAddressProvider")
+    void getUnionAddress_whenAddressSetAndStored_shouldReturnSetAddress(RskAddress rskAddress) {
+        // Arrange
+        unionStorageProvider.setUnionAddress(rskAddress);
+        unionStorageProvider.save();
+
+        // Act
+        Optional<RskAddress> unionBridgeAddress = unionStorageProvider.getUnionAddress();
+
+        // Assert
+        assertTrue(unionBridgeAddress.isPresent());
+        assertEquals(rskAddress, unionBridgeAddress.get());
+    }
+
+    @Test
+    void setUnionAddress_whenAddressSet_shouldNotStore() {
+        // Arrange
+        RskAddress rskAddress = TestUtils.generateAddress("address");
+
+        // Act
+        unionStorageProvider.setUnionAddress(rskAddress);
+
+        // Assert
+        verify(mockStorageAccessor, never())
+            .saveToRepository(any(), any(), any());
+    }
+
+    @Test
+    void save_whenNoAddressSet_shouldNotStoreAnything() {
+        // Act
+        unionStorageProviderUsingMockStorage.save();
+
+        // Assert
+        verify(mockStorageAccessor, never())
+            .saveToRepository(any(), any(), any());
+    }
+
+    @ParameterizedTest
+    @MethodSource("validAddressProvider")
+    void getAndSetUnionAddress_whenFirstTime_ok(RskAddress rskAddress) {
+        // Act & Assert
+        // Check that the address is not present in the storage nor in cache initially
+        Optional<RskAddress> unionBridgeAddress = unionStorageProvider.getUnionAddress();
+        assertTrue(unionBridgeAddress.isEmpty());
+
+        // Set the address
+        unionStorageProvider.setUnionAddress(rskAddress);
+
+        // Check that the address is now present in the cache
+        Optional<RskAddress> cachedAddress = unionStorageProvider.getUnionAddress();
+        assertTrue(cachedAddress.isPresent());
+        assertEquals(rskAddress, cachedAddress.get());
+
+        // Check that the address is now present in the storage
```

Comment:
```suggestion
        // Save the value
```
you are not checking anything in that line right? Just saving the value

Reply:
Updated